master/buildbot/steps/artifact.py

```diff
@@ -379,6 +448,108 @@ def start(self):
         self.setCommand(command)
         ShellCommand.start(self)
 
+    @defer.inlineCallbacks
+    def _getBuildRequest(self):
+        triggeredbybrid = self.build.requests[0].id
+        br = yield self.master.db.buildrequests.getBuildRequestTriggered(triggeredbybrid, self.artifactBuilderName)
+        defer.returnValue(br)
+
+
+class DownloadArtifactFromParent(DownloadArtifact):
+    @defer.inlineCallbacks
+    def _getBuildRequest(self):
+        triggeredbybrid = self.build.requests[0].id
+        id = yield self.master.db.buildrequests.getTriggeredById(triggeredbybrid)
+        br = yield self.master.db.buildrequests.getBuildRequestById(id)
+        defer.returnValue(br)
+
+class DownloadArtifactFromChilden(LoggingBuildStep, CompositeStepMixin):
+    def __init__(self, name=None, projectPrefix=None, targetConfig=None, artifactBuilderName=None,
+                 artifact=None, artifactDirectory=None, artifactDestination=None,
+                 artifactServer=None, artifactServerDir=None, artifactServerPort=None,
+                 usePowerShell=True, baseLocalDir=None, **kwargs):
+        self.workdir = 'build'
+        self.artifactBuilderName = artifactBuilderName
+        self.artifact = artifact
+        self.projectPrefix = projectPrefix
+        self.target_config = targetConfig
+        self.artifactDirectory = artifactDirectory
+        self.artifactServer = artifactServer
+        self.artifactServerDir = artifactServerDir
+        self.artifactServerPort = artifactServerPort
+        self.artifactDestination = artifactDestination or artifact
+        self.master = None
+        self.usePowerShell = usePowerShell
+        self.kwargs = kwargs
+        self.name = name
```

Comment:
`name`, `description` and `descriptionDone` should be automatic generated or just constants on the class